### PR TITLE
[FLINK-25880][Docs] Implement Matomo in Flink documentation

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* This PR removes the current Google Analytics web analytics tracking from the Flink documentation and replaces it with the ASF hosted Matomo installation, as discussed on the Dev mailing list https://lists.apache.org/thread/kv0mf1y9cq4cdsf4jkq3gb6jo7b81cq3

## Brief change log

* ef12918d1aa00b4c529f34a5a7ed6a1edd79c920 Removes Google Analytics implementation
* 0b1ad7917d4839b0226cb874702ed4089a91e706 Adds Matomo setup. This was previously done for flink-web via https://github.com/apache/flink-web/pull/502 and https://github.com/apache/flink-web/pull/503

## Verifying this change

* You can inspect the network traffic in your browser using the Developer Tools. Right now, you'll see calls being made to the google-analytics.com domain. When checking ef12918d1aa00b4c529f34a5a7ed6a1edd79c920 you'll see that these calls are no longer made.

*  0b1ad7917d4839b0226cb874702ed4089a91e706 introduces calls being made to https://matomo.privacy.apache.org/ - This is where the self-hosted Matomo instance is hosted (on a VM provided by ASF Infra)

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable / the goal and what data is stored in Matomo is explained in the Privacy Policy on the Flink project website.

### CODE REVIEW

File: docs/layouts/_default/baseof.html [hunk 1]
```diff
@@ -23,6 +23,25 @@
...
   {{ partial "docs/html-head" . }}
   {{ partial "docs/inject/head" . }}
+    <!-- Matomo -->
+    <script>
+      var _paq = window._paq = window._paq || [];
+      /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
+      /* We explicitly disable cookie tracking to avoid privacy issues */
+      _paq.push(['disableCookies']);
+      /* Measure a visit to flink.apache.org and nightlies.apache.org/flink as the same visit */
+      _paq.push(["setDomains", ["*.flink.apache.org","*.nightlies.apache.org/flink"]]);
```
Comment (knaufk) [inline docs/layouts/_default/baseof.html:33]: Isn't the documentation always under nightlies?

Comment (MartijnVisser) [reply docs/layouts/_default/baseof.html:33]: The documentation is, but the project website isn't. By setting this value, we make sure that when a visit is starting at flink.apache.org and then clicks on documentation (and vice versa) is tracked as the same visit and not starting a new one. Else each click would appear as an 'outlink' like the user left our website, which isn't the case.

Comment (knaufk) [reply docs/layouts/_default/baseof.html:33]: Understood.

**Review by knaufk — APPROVED** [review]
LGTM. This needs to be backported, right? Will you create PRs for those, too?

**Comment by MartijnVisser** [comment]
> LGTM. This needs to be backported, right? Will you create PRs for those, too?

Yes, this needs to be backported and I'll create those.

**Pull request merged**